searx/plugins/hostname_replace.py

```diff
@@ -0,0 +1,28 @@
+import re
+from urllib.parse import urlunparse
+from searx import settings
+from searx.plugins import logger
+from flask_babel import gettext
+
+name = gettext('Hostname replace')
+description = gettext('Rewrite result hostnames')
+default_on = False
+preference_section = 'general'
+
+plugin_id = 'hostname_replace'
+parsed = 'parsed_url'
+
+replacements = {re.compile(p): r for (p, r) in settings[plugin_id].items()} if plugin_id in settings else {}
+
+logger = logger.getChild(plugin_id)
+
+
+def on_result(request, search, result):
+    if parsed not in result:
+        return True
+    for (pattern, replacement) in replacements.items():
+        if pattern.search(result[parsed].netloc):
+            result[parsed] = result[parsed]._replace(netloc=pattern.sub(replacement, result[parsed].netloc))
```

Comment:
I did not know, that `._replace()` is an [documented method of named tuples](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace).  Should we use it here / I don't now.

Reply:
Do you see any reason not to?
FWIW it's already in use by `tracker_url_remover`, and in `results.py`